cirrus/google_cloud/iam.py

```diff
@@ -0,0 +1,166 @@
+"""
+Google IAM Helper Classes and Functions
+"""
+
+
+class GooglePolicy(object):
+    """
+    A Google Policy with bindings between members and roles
+    """
+
+    def __init__(self, bindings, etag="", version=0):
+        """
+        Constructs a Google Policy
+
+        Args:
+            bindings (List(GooglePolicyBinding)): Connections between members and roles
+            etag (str): etag is used for optimistic concurrency control as a way to help
+                        prevent simultaneous updates of a policy from overwriting each other
+                         - Google
+            version (int): version for the policy
+        """
+        self.bindings = bindings
+        self.etag = etag
+        self.version = version
+
+    def __repr__(self):
```

Comment:
Nitpick: `__repr__` is used specifically to give a unique representation of the python object (which is why the return from `__repr__` generally looks something like `'<__main__.Foo object at 0x103816810>'`); `__str__` is probably more fitting.

Reply:
I agree, got in the habit of creating `__repr__` instead of `__str__` for everything for dual purpose of printline debugging and seeing it nicely in a debugger. I think `__repr__` could also be a way to recreate the object (e.g. something like `MyClass("arg1", 2)`). Anyway, I agree, `__str__` is more fitting in this case